dealiiadapter.cc

```diff
@@ -880,83 +1019,93 @@ void ElasticProblem<dim>::initialize_precice()
+                    for(int jj=0; jj<dim; ++jj)
+                        interface_faces_positions[face_position_iterator * dim + jj] = cell->face(face)->center()[jj];
 
-                //locally_owned_face_indices[coupling_face_iter]=cell->face_index(f);
-                //store the index of the coupling face
-                //this is necessary to assign the right face with the right value using multithreading
-                ++face_position_iterator;
+                    ++face_position_iterator;
+                }
             }
-        }
 
 
-    // pass face coordinates to precice
-    precice.setMeshVertices(face_mesh_id, n_interface_faces, interface_faces_positions.data(), interface_faces_ids.data());
+        // pass face coordinates to precice
+        precice.setMeshVertices(face_mesh_id, n_interface_faces, interface_faces_positions.data(), interface_faces_ids.data());
 
-    precice.initialize();
+        precice.initialize();
 
-    //Write initial writeData to preCICE
-    if (precice.isActionRequired(precice::constants::actionWriteInitialData()))
-    {
-        // store initial write_data for precice in precice_displacements
-        extract_relevant_displacements(precice_displacements);
+        //Write initial writeData to preCICE
+        if (precice.isActionRequired(precice::constants::actionWriteInitialData()))
+        {
+            // store initial write_data for precice in precice_displacements
+            extract_relevant_displacements(precice_displacements);
 
-        precice.writeBlockVectorData(displacements_data_id, n_interface_nodes, interface_nodes_ids.data(), precice_displacements.data());
-        precice.fulfilledAction(precice::constants::actionWriteInitialData());
+            precice.writeBlockVectorData(displacements_data_id, n_interface_nodes, interface_nodes_ids.data(), precice_displacements.data());
+            precice.fulfilledAction(precice::constants::actionWriteInitialData());
 
-        precice.initializeData();
-    }
+            precice.initializeData();
+        }
 
 
-    //Read initial readData from preCICE for the first time step
-    if (precice.isReadDataAvailable())
-        precice.readBlockVectorData(forces_data_id, n_interface_faces, interface_faces_ids.data(), precice_forces.data());
+        //Read initial readData from preCICE for the first time step
+        if (precice.isReadDataAvailable())
+            precice.readBlockVectorData(forces_data_id, n_interface_faces, interface_faces_ids.data(), precice_forces.data());
 
-    // this function is currently unnecessary (and wrong), since the precice forces are directly used in the assembly function
-    //    apply_precice_forces(precice_forces);
+        // this function is currently unnecessary (and wrong), since the precice forces are directly used in the assembly function
+        //    apply_precice_forces(precice_forces);
```

Comment:
If unnecessary please remove

Reply:
I could remove it. But maybe we should keep it from a conceptual point of view. People could see in the `advance()` function, that deal.II gets the read data. Currently, it is directly used in the assembly.